writers/streamingbatchwriter/streamingbatchwriter_test.go

```diff
@@ -379,9 +379,10 @@ func TestStreamingBatchUpserts(t *testing.T) {
 func waitForLength(t *testing.T, checkLen func(messageType) int, msgType messageType, want int) {
 	t.Helper()
 	lastValue := -1
+	timeout := time.After(5 * time.Second)
 	for {
 		select {
-		case <-time.After(time.Second):
+		case <-timeout:
 			t.Fatalf("timed out waiting for %v message length %d (last value: %d)", msgType, want, lastValue)
 		default:
```

Comment:
There was a bug here that caused the test to hang indefinitely; moving the timeout up solves it